src/univariate/continuous/pgeneralizedgaussian.jl

```diff
@@ -0,0 +1,110 @@
+
+"""
+    PGeneralizedGaussian(α, μ, p)
+
+The *p-Generalized Gaussian distribution*, more commonly known as the exponential
+power or the generalized normal distribution, with scale 'α', location 'μ', and
+shape 'p' has the probability density function
+
+```math
+f(x, \\mu, \\alpha, p) = \\frac{p}{2\\alpha\\Gamma(1/p)} e^{-(\\frac{|x-\\mu|}{\\alpha})^p} \\quad x \\in (-\\infty, +\\infty) , \\alpha > 0, p > 0
+```
+
+The p-Generalized Gaussian (GGD) is a parametric distribution that incorporates the
+Normal and Laplacian distributions as special cases where p = 1 and p = 2. As p → ∞,
+the distribution approaches the Uniform distribution on (μ-α, μ+α).
+
+```julia
+PGeneralizedGaussian()           # GGD with shape 2, scale 1, location 0, (the Normal distribution)
+PGeneralizedGaussian(loc,s,sh)   # GGD with location loc, scale s, and shape sh
+
+params(d)                       # Get the parameters, i.e. (loc,s,sh,)
+shape(d)                        # Get the shape parameter, sh
+scale(d)                        # Get the scale parameter, s
+location(d)                     # Get the location parameter, loc
+```
+
+External Links
+ * [Generalized Gaussian on Wikipedia](http://en.wikipedia.org/wiki/Generalized_normal_distribution)
+ * [Reference implementation paper](https://www.researchgate.net/publication/254282790_Simulation_of_the_p-generalized_Gaussian_distribution)
+ """
+
+struct PGeneralizedGaussian{T1<:Real, T2<:Real, T3<:Real} <: ContinuousUnivariateDistribution
+    α::T1
+    μ::T2
+    p::T3
+
+    PGeneralizedGaussian(μ::T1,α::T2,p::T3) where {T1,T2,T3} = (@check_args(PGeneralizedGaussian, α > zero(α) && p > zero(p)); new{T1,T2,T3}(μ,α,p))
+end
+
+PGeneralizedGaussian(μ::T1,α::T2,p::T3) where {T1<:Real,T2<:Real,T3<:Real} = PGeneralizedGaussian{T1,T2,T3}(μ,α,p)
+PGeneralizedGaussian(μ::Integer,α::Integer,p::Integer) = PGeneralizedGaussian(Float64(μ),Float64(α),Float64(p))
+PGeneralizedGaussian(p::Float64) = PGeneralizedGaussian(0.0, 1.0, p)
```

Comment:
The two constructors with default values should have a docstring attached, and not force the parameter type:
```julia
"""
    PGeneralizedGaussian(p)

Builds a p-generalized gaussian with `µ=0.0, α=1.0`
"""
PGeneralizedGaussian(p::T) where {T <: Real} = PGeneralizedGaussian(0.0, 1.0, p)
```